### PR TITLE
Add NVTX markers that dump additional information for nvprim_nvfuser Dynamo graphs

### PR DESCRIPTION
**Pull request by blongnv**
dump information on graphs that NVFuser JIT compiles:
- the markers show the list of ops, args, and inputs that make up the graph

also dumps information on FX nodes that are not touched by NVFuser:
- the markers show the op, name, and arg list of the node

cc @kevinstephano @jjsjann123

### CODE REVIEW

File: torch/_prims/nvfuser_executor.py
Comment (IvanYashchuk) [inline torch/_prims/nvfuser_executor.py:227]: Let's save `os.getenv('PYTORCH_NVFUSER_DUMP_NVTX')` outside this function call, no need to query `os` for every execution of this function.

File: torch/_prims/nvfuser_executor.py
Comment (IvanYashchuk) [inline torch/_prims/nvfuser_executor.py:369]: kwargs are missing

File: torch/_prims/nvfuser_executor.py
Comment (IvanYashchuk) [inline torch/_prims/nvfuser_executor.py]: Did you want to keep `n.meta` or removing it is intentional?

Comment (blongnv) [reply torch/_prims/nvfuser_executor.py]: Intentionally removed it, that was only for a local experiment

**Review by IvanYashchuk — APPROVED** [review]
Looks good!

`range_push` and `range_pop` are fine, there's also `torch.cuda.nvtx.range` context manager, but [for some reason, it's not documented](https://pytorch.org/docs/stable/cuda.html#nvidia-tools-extension-nvtx).

File: torch/_prims/nvfuser_executor.py
Comment (jjsjann123) [inline torch/_prims/nvfuser_executor.py:224]: @kevinstephano Here's the NVTX marker.

File: torch/_prims/nvfuser_executor.py
Comment (jjsjann123) [inline torch/_prims/nvfuser_executor.py:32]: Let's do this in a function instead. This would allow us the set the env variable inside a user script.

e.g.
```
@lru_cache(None)
def get_nvprim_dump_nvtx():
  return os.getenv("PYTORCH_NVFUSER_DUMP_NVTX")
```

**Review by jjsjann123 — APPROVED** [review]
LGTM, a quick comment on the getenv.

File: torch/_prims/nvfuser_executor.py
Comment (blongnv) [reply torch/_prims/nvfuser_executor.py:32]: smart, will change this

**Comment by jjsjann123** [comment]
Looks like we are happy with the code change. I'm pulling Ruberry with the merge button to stamp.

**Comment by mruberry** [comment]
Lint failures look real -- are the other CUDA failures related?

**Comment by jjsjann123** [comment]
Pretty sure failures are just flaky, since all tests passed prior to merging the last lint patches. restarted the test.

Meanwhile, cc'ing @mruberry for review/stamp

**Comment by jjsjann123** [comment]
CI all green~~ :tada: 

Now we just need a stamp before we can command the bot to merge it.

**Review by mruberry — APPROVED** [review]
Stamped!

**Pull request closed without merging**
